Migrate 4D vertexing and downstream reco to full simulation/reconstruction track timestamps and add TOF PID (10_5_X)

bendavid opened the pull request:
(master version of https://github.com/cms-sw/cmssw/pull/25627)
Implements changes needed to propagate full simulation and reconstruction based track timestamps to 4D vertex reconstruction and particle flow.
(Previously this was already implemented and running based on the fastsim timestamps used for the MTD Technical Proposal)

The following strategy is used to deal with particles moving slower than the speed of light:

Existing timestamps from the TrackExtender assume the pion mass hypothesis when propagating the MTD cluster time back to the beamline. The valuemaps produced at this step have been extended to provide some additional information in a more convenient form. (Time at MTD, momentum magnitude of refitted track, etc)  As well the uncertainty is inflated by adding in quadrature the time of flight difference between the pion and proton mass hypothesis for each track.

First pass of 4D vertex reconstruction runs on the output as above. After a bugfix/protection, combined with the inflated uncertainties it is able to deal with the resulting outliers from slower kaons/protons in a reasonable way.

Additional TOFPIDProducer which tests compatibility of kaon and proton mass hypotheses for the tracks with the 4D vertices from 2) and produces updated timestamps where appropriate. This producer also creates value maps of particle ID probabilities which can be used downstream.

Second pass of 4D vertex reconstruction based on the refined timestamps from 3)

This pull request also migrates the PFCandidate timestamps to the fullsim/reconstruction ones (previously populated from the fastsim). This automatically propagates to the packed candidates in MINIAOD as well.

An additional collection of slimmed primary vertices with timing is added to MINIAOD, but the PackedCandidate-vertex association, puppi, etc are still based on the default primary vertices without timing.

In order to facilitate comparisons and benchmarking, the fastsim-based timestamps and 4d vertex reco is still run and stored. This is especially relevant since direct comparisons may validate the continued use of existing/fastsim-based results where fullsim-based ones are not yet available.

Some discussion and validation/performance plots in https://indico.cern.ch/event/783100/contributions/3273442/attachments/1780769/2896933/mtdvtxJan17-2019.pdf



@pmeridian @lgray

bendavid committed:
migrate 4d vertexing to fullsim and first mostly working version of time of flight particle id

bendavid committed:
add protection for infinite loop in 4d vertex reco

bendavid committed:
migrate PFCandidate times to fullsim/reconstruction

bendavid committed:
produce also fastsim-based 4d vertices for now

bendavid committed:
tweak pid algorithm and add some configurable parameters

bendavid committed:
rearrange RecoVertex configuration for timing era so that 4d vertices populate the default offlinePrimaryVertices collections

cmsbuild commented:
The code-checks are being triggered in jenkins.

kpedro88 commented:
assign upgrade

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25628/7942

- This PR adds an extra 44KB to repository

cmsbuild commented:
New categories assigned: upgrade

@kpedro88 you have been requested to review this Pull request/Issue and eventually sign? Thanks

cmsbuild commented:
A new Pull Request was created by  @bendavid (Josh Bendavid) for master.

It involves the following packages:

CommonTools/RecoAlgos
RecoMTD/TrackExtender
RecoParticleFlow/PFProducer
RecoVertex/Configuration
RecoVertex/PrimaryVertexProducer


@perrotta, @cmsbuild, @kpedro88, @slava77 can you please review it and eventually sign? Thanks.
@rappoccio, @jdolen, @makortel, @abbiendi, @GiacomoSguazzoni, @mmarionncern, @jhgoh, @lgray, @ahinzmann, @gkasieczka, @bachtis, @cbernet, @ebrondol, @VinInn, @rovere, @dgulhan, @seemasharmafnal, @clelange this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: f487a29343e52b03fabc389b12228816f9a43647

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25628/32538/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found compilation warning when building:  See details on the summary page.

* **Clang**:

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 16 COMPILER='llvm compile'

```
 See details on the summary page.

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

slava77 commented on CommonTools/RecoAlgos/plugins/TOFPIDProducer.cc:
```diff
@@ -0,0 +1,314 @@
+  TOFPIDProducer(const ParameterSet& pset); 
+
+  void produce(edm::Event& ev, const edm::EventSetup& es) final;
+
```
new modules should have a ```fillDescriptions``` method defined

slava77 commented on CommonTools/RecoAlgos/python/TOFPIDProducer_cfi.py:
```diff
@@ -0,0 +1,16 @@
+import FWCore.ParameterSet.Config as cms
+
+TOFPIDProducer = cms.EDProducer( "TOFPIDProducer",
```
the instance better follow the lower case convention -> ```tofPIDProducer```.
Perhaps even ```tofPID``` is good enough: we typically drop the "Producer" from the collection names

slava77 commented on RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc:
```diff
@@ -288,22 +314,30 @@ void TrackExtenderWithMTDT<TrackCollection>::produce( edm::Event& ev,
+    sigmatmtdMap = sigmatmtd;
         auto& backtrack = output->back();
+    pMap = backtrack.p();
 	betaMap = backtrack.beta();
```
indentation

slava77 commented on RecoParticleFlow/PFProducer/python/particleFlowBlock_cfi.py:
```diff
@@ -179,10 +179,10 @@ def _findIndicesByModule(name):
-                             timeErrorMapGsf = cms.InputTag("gsfTrackTimeValueMapProducer:electronGsfTracksConfigurableFlatResolutionModelResolution")
+                             timeValueMap = cms.InputTag("TOFPIDProducer:t0"),
+                             timeErrorMap = cms.InputTag("TOFPIDProducer:sigmat0"),
+                             timeValueMapGsf = cms.InputTag("TOFPIDProducer:t0"), #this will cause no time to be set for gsf tracks (since this is not available for the fullsim/reconstruction yet) *TODO* update when gsf times are available
```
newlne, please or add the comment above in multiple lines

slava77 commented on RecoParticleFlow/PFProducer/python/simPFProducer_cfi.py:
```diff
@@ -19,10 +19,10 @@
-    gsfTrackTimeErrorMap = cms.InputTag("gsfTrackTimeValueMapProducer:electronGsfTracksConfigurableFlatResolutionModelResolution"),
+    trackTimeValueMap = cms.InputTag("TOFPIDProducer:t0"),
+    trackTimeErrorMap = cms.InputTag("TOFPIDProducer:sigmat0"),
+    gsfTrackTimeValueMap = cms.InputTag("TOFPIDProducer:t0"), #this will cause no time to be set for gsf tracks (since this is not available for the fullsim/reconstruction yet) *TODO* update when gsf times are available
```
newline(s)

slava77 commented on RecoVertex/Configuration/python/RecoVertex_cff.py:
```diff
@@ -36,33 +36,78 @@
+phase2_timing.toReplaceWith(offlinePrimaryVertices, offlinePrimaryVertices4D)
+phase2_timing.toReplaceWith(offlinePrimaryVerticesWithBS, offlinePrimaryVertices4DWithBS)
+phase2_timing.toModify(offlinePrimaryVertices, vertices = cms.InputTag("unsortedOfflinePrimaryVertices"), particles = cms.InputTag("trackRefsForJetsBeforeSorting"))
+phase2_timing.toModify(offlinePrimaryVerticesWithBS, vertices = cms.InputTag("unsortedOfflinePrimaryVertices","WithBS"), particles = cms.InputTag("trackRefsForJetsBeforeSorting"))
```
please drop type specifications for all parameters which already exist.
This is a safer way to protect from parameter name mistakes and will also help in possible parameter migrations.

slava77 commented on RecoVertex/Configuration/python/RecoVertex_cff.py:
```diff
@@ -36,33 +36,78 @@
+trackWithVertexRefSelectorBeforeSorting4DnoPID.ptErrorCut=9e99
+trackRefsForJetsBeforeSorting4DnoPID = trackRefsForJets.clone(src="trackWithVertexRefSelectorBeforeSorting4DnoPID")
+offlinePrimaryVertices4DnoPID=sortedPrimaryVertices.clone(vertices="unsortedOfflinePrimaryVertices4DnoPID", particles="trackRefsForJetsBeforeSorting4DnoPID", trackTimeTag=cms.InputTag("trackExtenderWithMTD","generalTrackt0"),trackTimeResoTag=cms.InputTag("trackExtenderWithMTD","generalTracksigmat0"),assignment=dict(useTiming=True))
+offlinePrimaryVertices4DnoPIDWithBS=sortedPrimaryVertices.clone(vertices="unsortedOfflinePrimaryVertices4DnoPID:WithBS", particles="trackRefsForJetsBeforeSorting4DnoPID", trackTimeTag=cms.InputTag("trackExtenderWithMTD","generalTrackt0"),trackTimeResoTag=cms.InputTag("trackExtenderWithMTD","generalTracksigmat0"),assignment=dict(useTiming=True))
```
newlines

It seems also practical to move all the phase-2 timing vertices to their own file.

slava77 reviewed (commented):
this is just a quick visual check.
I'll have a more careful look once the PR tests by jenkins can actually complete.

bendavid committed:
remove beamspot from TOFPIDProducer (not used for the moment)

cmsbuild commented:
The code-checks are being triggered in jenkins.

bendavid commented:
(that should fix compilation warnings at least)

slava77 commented:
@cmsbuild please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25628/7947

- This PR adds an extra 44KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #25628 was updated. @perrotta, @cmsbuild, @kpedro88, @slava77 can you please check and sign again.

cmsbuild commented:
-1

Tested at: cbda9a7e05f3e63a3b9fd4a8b1bd81f4fc6db948

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25628/32543/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
27434.0 step3
<pre>runTheMatrix-results/27434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D35+RecoFullGlobal_2023D35+HARVESTFullGlobal_2023D35/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D35+RecoFullGlobal_2023D35+HARVESTFullGlobal_2023D35.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

bendavid committed:
fix input collections for validation sequence

bendavid commented:
Sorry, I had not tested the validation sequence.  Input collections there should be fixed now as well.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25628/7960

- This PR adds an extra 56KB to repository

cmsbuild commented:
Pull request #25628 was updated. @perrotta, @andrius-k, @kmaeshima, @schneiml, @kpedro88, @cmsbuild, @jfernan2, @slava77 can you please check and sign again.

perrotta commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: f7cade4504967eb876981b0296efe9d7c067f8d3
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25628/32553/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25628/32553/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 736 differences found in the comparisons
 - DQMHistoTests: Total files compared: 33
 - DQMHistoTests: Total histograms compared: 3152865
 - DQMHistoTests: Total failures: 994
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3151667
 - DQMHistoTests: Total skipped: 204
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 32 files compared)
 - Checked 137 log files, 14 edm output root files, 33 DQM output files

slava77 commented:
@bendavid 
please provide a comparison of CPU timing with PU200

pmeridian commented:
These are the numbers that I get on 100 ttbar events:
TimeReport ---------- Event  Summary ---[sec]----
TimeReport       event loop CPU/event = 853.101740
TimeReport      event loop Real/event = 226.080010
TimeReport     sum Streams Real/event = 889.604837
TimeReport efficiency CPU/Real/thread = 0.943363

In particular:
TimeReport 287.662276   287.662276   287.662276  unsortedOfflinePrimaryVertices
TimeReport   9.830491     9.830491     9.830491  unsortedOfflinePrimaryVertices3D
TimeReport  22.177901    22.177901    22.177901  unsortedOfflinePrimaryVertices4Dfastsim
TimeReport 377.122103   377.122103   377.122103  unsortedOfflinePrimaryVertices4DnoPID

slava77 commented:
On 1/14/19 7:19 AM, pmeridian wrote:
> These are the numbers that I get on 100 ttbar events:
> TimeReport ---------- Event Summary ---[sec]----
> TimeReport event loop CPU/event = 853.101740
> TimeReport event loop Real/event = 226.080010
> TimeReport sum Streams Real/event = 889.604837
> TimeReport efficiency CPU/Real/thread = 0.943363
> 
> In particular:
> TimeReport 287.662276 287.662276 287.662276 unsortedOfflinePrimaryVertices
> TimeReport 9.830491 9.830491 9.830491 unsortedOfflinePrimaryVertices3D
> TimeReport 22.177901 22.177901 22.177901
> unsortedOfflinePrimaryVertices4Dfastsim
> TimeReport 377.122103 377.122103 377.122103
> unsortedOfflinePrimaryVertices4DnoPID
> 
> —

@pmeridian
do you have the same numbers from the baseline version (without changes
from this PR)?

pmeridian commented:
TimeReport ---------- Event  Summary ---[sec]----
TimeReport       event loop CPU/event = 188.674736
TimeReport      event loop Real/event = 59.912504
TimeReport     sum Streams Real/event = 231.298837
TimeReport efficiency CPU/Real/thread = 0.787293

TimeReport   9.881747     9.881747     9.881747  unsortedOfflinePrimaryVertices
TimeReport  22.361680    22.361680    22.361680  unsortedOfflinePrimaryVertices4D

slava77 commented:
@bendavid 
please clarify on the status of the follow up to the code review
https://github.com/cms-sw/cmssw/pull/25628#pullrequestreview-191870782

bendavid commented:
Will take care of this in the next few hours (while considering possible mitigations for the excessive cpu usage evidently driven by tracks with outlier timestamps)

kpedro88 commented on CommonTools/RecoAlgos/plugins/TOFPIDProducer.cc:
```diff
@@ -0,0 +1,308 @@
+
+using namespace std;
+using namespace edm;
+
```
remove extra whitespace

kpedro88 commented on CommonTools/RecoAlgos/plugins/TOFPIDProducer.cc:
```diff
@@ -0,0 +1,308 @@
+
+
+class TOFPIDProducer : public edm::stream::EDProducer<> {  
+  static constexpr char t0Name[] = "t0";
```
code organization: private members should go in the explicitly-labeled "private" section

kpedro88 commented on CommonTools/RecoAlgos/plugins/TOFPIDProducer.cc:
```diff
@@ -0,0 +1,308 @@
+  
+  constexpr double m_k = 0.493677; //[GeV]
+  constexpr double m_p = 0.9382720813; //[GeV]
+  constexpr double c = 2.99792458e1; //[cm/ns]
```
better to get speed of light from e.g. CLHEP

kpedro88 commented on CommonTools/RecoAlgos/plugins/TOFPIDProducer.cc:
```diff
@@ -0,0 +1,308 @@
+    probPOutRaw.push_back(prob_p);
+  }
+
+  edm::ValueMap<float>::Filler fillert0s(*t0Out);
```
code repetition could be reduced by defining a helper function to fill the value maps (see also https://github.com/cms-sw/cmssw/pull/25063/files#r241820150)

kpedro88 commented on RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc:
```diff
@@ -61,10 +61,15 @@ using namespace edm;
 template<class TrackCollection>
 class TrackExtenderWithMTDT : public edm::stream::EDProducer<> {  
   static constexpr char pathLengthName[] = "pathLength";
```
same comment about organization of private members

kpedro88 commented on RecoVertex/Configuration/python/RecoVertex_cff.py:
```diff
@@ -36,33 +36,78 @@
+unsortedOfflinePrimaryVertices4DnoPID.TrackTimesLabel = cms.InputTag("trackExtenderWithMTD:generalTrackt0")
+unsortedOfflinePrimaryVertices4DnoPID.TrackTimeResosLabel = cms.InputTag("trackExtenderWithMTD:generalTracksigmat0")
+trackWithVertexRefSelectorBeforeSorting4DnoPID = trackWithVertexRefSelector.clone(vertexTag="unsortedOfflinePrimaryVertices4DnoPID")
+trackWithVertexRefSelectorBeforeSorting4DnoPID.ptMax=9e99
```
parameters can be changed within the clone statement, rather than afterward (reduce repetition of module name)

kpedro88 commented on RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT_vect.cc:
```diff
@@ -676,10 +676,10 @@ DAClusterizerInZT_vect::split(const double beta,  track_t &tks, vertex_t & y, do
-    while(   (find_nearest(z1, t1, y, k_min1, epsilonz, epsilont) && (k_min1 != k))
-	     || (find_nearest(z2,t2, y, k_min2, epsilonz, epsilont) && (k_min2 !=k)) ){
+    while(   ((find_nearest(z1, t1, y, k_min1, epsilonz, epsilont) && (k_min1 != k))
+	     || (find_nearest(z2,t2, y, k_min2, epsilonz, epsilont) && (k_min2 !=k))) && (std::abs(z2-z1)>1e-8 || std::abs(t2-t1)>1e-8) ){
```
magic number 1e-8 should be named constant

slava77 commented:
> Will take care of this in the next few hours (while considering possible mitigations for the excessive cpu usage evidently driven by tracks with outlier timestamps)

Did the outlook for update become more clear?
I'm not sure I'll be able to look at this later today if it is still updated today.

bendavid commented:
We've been focusing on solving the CPU time issue.  It looks like we have a viable solution and can update the PR later tonight.

At that point this PR will add something like 30s/event to the reco time for ttbar pu200 (as opposed to 700s now)

bendavid committed:
fix computation of vertex time uncertainty

bendavid committed:
add outlier rejection for use of timing information in 4d vertex reconstruction

bendavid committed:
inflate uncertainty on track times to account for ambiguity in mass hypothesis at low momentum

bendavid committed:
update logic and settings in TOFPIDProducer to produce an additional safe t0 and sigmat0 output which protects the downstream 4d vertex reconstruction from outliers by keeping inflated uncertainties for tracks with ambiguous particle id

bendavid committed:
code cleanup and review

bendavid committed:
add primary vertex configuration for mtd to separate file

bendavid committed:
add fillDescriptions to TOFPIDProducer

bendavid committed:
add helper functions to fill value maps

bendavid committed:
make outlier rejection remove tracks entirely rather than just drop timing info

bendavid committed:
tune parameters for particle id

bendavid committed:
further improvements to vertex time uncertainty calculation

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25628/8043

- This PR adds an extra 116KB to repository

cmsbuild commented:
Pull request #25628 was updated. @perrotta, @andrius-k, @kmaeshima, @schneiml, @kpedro88, @cmsbuild, @jfernan2, @slava77 can you please check and sign again.

bendavid commented:
Addressed all review comments from @slava77 and @kpedro88 and added some fixes/improvements which radically reduce the CPU consumption of the 4D vertex reconstruction by
1) Removing extreme outliers in time
2) For the first pass before PID has been applied, inflate the time uncertainty of the tracks with the full difference from pion vs proton mass hypothesis
3) From the PID step, produce an additional set of "safe" value maps for use by the final 4D vertex step, in which tracks with ambiguous particle ID retain the inflated time uncertainty

This all has the effect of either removing outliers in time or inflating their uncertainty such that they no longer provoke bad behaviour in the algorithm.

Future improvements/protections are possible in the 4D vertex reco itself, and both the vertex reco and particle ID can likely be improved by moving to an iterative algorithm in the future.

Performance numbers for PU200 coming shortly.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

bendavid commented:
For TTBar PU200 100 events:
TimeReport> Time report complete in 888.779 seconds
 Time Summary: 
 - Min event:   51.5563
 - Max event:   280.765
 - Avg event:   125.223
 - Total loop:  848.592
 - Total init:  37.7146
 - Total job:   888.779
 - EventSetup Lock:   551.946
 - EventSetup Get:   116.436
 Event Throughput: 0.117842 ev/s
 CPU Summary: 
 - Total loop:  12410.8
 - Total init:  17.2199
 - Total extra: 0
 - Total job:   12435.4
 Processing Summary: 
 - Number of Events:  100
 - Number of Global Begin Lumi Calls:  1
 - Number of Global Begin Run Calls: 1

and in particular
unsortedOfflinePrimaryVertices 13.77
unsortedOfflinePrimaryVertices4DnoPID 7.91
unsortedOfflinePrimaryVertices4Dfastsim 16.56
unsortedOfflinePrimaryVertices3D 5.18
tofPID 0.045210270999999976
trackExtenderWithMTD 1.7737585000000005
trackTimeValueMapProducer 0.07095373099999999

So the additional 4D vertex producers are now adding a little over 20s/event.
(In the future we can drop unsortedOfflinePrimaryVertices4Dfastsim and trackTimeValueMapProducer, but for now we prefer to keep them for debugging purposes and 1:1 comparison with what was used for the technical proposal)

cmsbuild commented:
+1
Tested at: 9cec85fe00a66876368b0f452dc4d9f56286cbdf
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25628/32653/summary.html

cmsbuild commented:
Comparison job queued.

andrius-k commented:
+1

bendavid commented:
Some discussion and validation/performance plots in https://indico.cern.ch/event/783100/contributions/3273442/attachments/1780769/2896933/mtdvtxJan17-2019.pdf

slava77 commented:
On 1/17/19 9:08 AM, Josh Bendavid wrote:
> Some discussion and validation/performance plots in
> https://indico.cern.ch/event/783100/contributions/3273442/attachments/1780769/2896933/mtdvtxJan17-2019.pdf
> 
> —

please add this to the PR description so that it's easier to find.
Thanks.

slava77 commented on RecoVertex/Configuration/python/RecoVertex_phase2_timing_cff.py:
```diff
@@ -0,0 +1,106 @@
+phase2_timing.toReplaceWith(vertexrecoTask, _phase2_tktiming_vertexrecoTask)
+phase2_timing.toReplaceWith(unsortedOfflinePrimaryVertices, unsortedOfflinePrimaryVertices4D)
+phase2_timing.toReplaceWith(offlinePrimaryVertices, offlinePrimaryVertices4D)
+phase2_timing.toReplaceWith(offlinePrimaryVerticesWithBS, offlinePrimaryVertices4DWithBS)
```
```phase2_timing``` is generic for all variants with timing.
Is this replacement fully applicable?
Shouldn't it be changed to phase2_timing_layer ?

bendavid replied:
So phase2_timing without phase2_timing_layer corresponds to the hybrid fast simulation based partly on simulation truth which was used for the MTD Technical Proposal.  Probably this will never be used again going forward, and I don't think we want to maintain it as such.  I'll find a minimal/clean solution here as not to break anything, since there are a few relvals defined which do use it...

slava77 replied:
thanks.
I checked locally that all of the non-```timing_layer``` wfs with timing are failing

```
	Era_Phase2C4_timing	= (Phase2C4, phase2_timing)	24034.11 D28 24834.11 D30 
	Era_Phase2C6_timing	= (Phase2C6, phase2_timing)	26234.11 D31 
	Era_Phase2_timing		= (Phase2, phase2_timing)	20034.11 D17 21234.11 D21 21634.11 D22 22034.11 D23 23234.11 D36 23634.11 D37
												24434.11 D29 
```

bendavid replied:
I've made sure that phase2+timing remains functional on its own for the moment.  We can remove it later (makes sense to do so at the moment where we fully remove the fastsim also from the phase2_timing_layer era, since we're keeping it in parallel now for comparisons)

bendavid commented on RecoVertex/Configuration/python/RecoVertex_phase2_timing_cff.py:
```diff
@@ -0,0 +1,106 @@
+unsortedOfflinePrimaryVertices4DnoPID = unsortedOfflinePrimaryVertices.clone( verbose = cms.untracked.bool(False),
+                                                                         TkClusParameters = DA2D_vectParameters,
+                                                                         TrackTimesLabel = cms.InputTag("trackExtenderWithMTD:generalTrackt0"),
+                                                                         TrackTimeResosLabel = cms.InputTag("trackExtenderWithMTD:generalTracksigmat0"),
```
The TrackTimesLabel and TrackTimesResosLabel parameters do not already exist in the module being cloned.

slava77 replied:
OK here, but please check/confirm for all other cases.

bendavid replied:
Confirmed that all of the remaining type specifications are for parameters that do not already exist.

bendavid commented on CommonTools/RecoAlgos/plugins/TOFPIDProducer.cc:
```diff
@@ -0,0 +1,355 @@
+        
+        float tmtd = tmtdIn[trackref];
+        float pathlength = pathLengthIn[trackref];
+        float magp = pIn[trackref];
```
Yes, the logic was just that the RHS is float.  Can do the type conversion already here, sure.

bendavid committed:
address further code review

bendavid committed:
make sure phase2_timing era (fastsim only for MTD) is still functional

cmsbuild commented:
The code-checks are being triggered in jenkins.

bendavid commented:
Addressed review comments and made sure phase2_timing era remains functional (checked that 20034.11 is working again)

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25628/8065

- This PR adds an extra 152KB to repository

- Found files with invalid states:
  - CommonTools/RecoAlgos/python/TOFPIDProducer_cfi.py:
    - Added: a0c1929358dfdfc8ff4a660084b0b4693f9a89e0
    - Modified: c834441618737eed29f1097cbfaf59430496c181, f487a29343e52b03fabc389b12228816f9a43647, cbda9a7e05f3e63a3b9fd4a8b1bd81f4fc6db948, b770cf63d1262306fa4e2bb237436d46ced5e583, f777755313b333cf770f408c3f05116ca7e6779f, 7810d5943e4b39736810665eaaf9d7f4d22f6007
    - Deleted: e1a553eff12d6d2f917ae4406f7c7d818fb2e7c2

cmsbuild commented:
Pull request #25628 was updated. @perrotta, @andrius-k, @kmaeshima, @schneiml, @kpedro88, @cmsbuild, @jfernan2, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test workflow 20034.11

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 0c7e8103880b66b4bd06f744439427b740f58665

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25628/32685/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals  AddOn**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
5.1 step1
<pre>runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS/step1_TTbar+TTbarFS+HARVESTFS.log

</pre>4.53 step3
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step3_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>136.731 step3
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step3_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>136.788 step3
<pre>runTheMatrix-results/136.788_RunSinglePh2017B+RunSinglePh2017B+HLTDR2_2017+RECODR2_2017reHLT_skimSinglePh_Prompt+HARVEST2017/step3_RunSinglePh2017B+RunSinglePh2017B+HLTDR2_2017+RECODR2_2017reHLT_skimSinglePh_Prompt+HARVEST2017.log

</pre>9.0 step3
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>25.0 step3
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step3_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>140.56 step2
<pre>runTheMatrix-results/140.56_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18/step2_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18.log

</pre>1306.0 step3
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step3_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>101.0 step1
<pre>runTheMatrix-results/101.0_SingleElectronE120EHCAL+SingleElectronE120EHCAL/step1_SingleElectronE120EHCAL+SingleElectronE120EHCAL.log

</pre>136.85 step3
<pre>runTheMatrix-results/136.85_RunEGamma2018A+RunEGamma2018A+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Prompt_L1TEgDQM+HARVEST2018_L1TEgDQM/step3_RunEGamma2018A+RunEGamma2018A+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Prompt_L1TEgDQM+HARVEST2018_L1TEgDQM.log

</pre>1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVDSIPIXELCALRUN1+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVDSIPIXELCALRUN1+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log

</pre>1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM.log

</pre>158.0 step3
<pre>runTheMatrix-results/158.0_HydjetQ_B12_5020GeV_2018_ppReco+HydjetQ_B12_5020GeV_2018_ppReco+DIGIHI2018PPRECO+RECOHI2018PPRECO+ALCARECOHI2018PPRECO+HARVESTHI2018PPRECO/step3_HydjetQ_B12_5020GeV_2018_ppReco+HydjetQ_B12_5020GeV_2018_ppReco+DIGIHI2018PPRECO+RECOHI2018PPRECO+ALCARECOHI2018PPRECO+HARVESTHI2018PPRECO.log

</pre>10042.0 step3
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017/step3_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017.log

</pre>10024.0 step3
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017.log

</pre>25202.0 step3
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step3_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>10824.0 step3
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018+NanoFull_2018/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+HARVESTFull_2018+ALCAFull_2018+NanoFull_2018.log

</pre>11624.0 step3
<pre>runTheMatrix-results/11624.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2019_GenSimFull+DigiFull_2019+RecoFull_2019+HARVESTFull_2019+ALCAFull_2019/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2019_GenSimFull+DigiFull_2019+RecoFull_2019+HARVESTFull_2019+ALCAFull_2019.log

</pre>10224.0 step3
<pre>runTheMatrix-results/10224.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017PU_GenSimFull+DigiFullPU_2017PU+RecoFullPU_2017PU+HARVESTFullPU_2017PU.log

</pre>20034.0 step3
<pre>runTheMatrix-results/20034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17.log

</pre>20034.11 step3
<pre>runTheMatrix-results/20034.11_TTbar_14TeV_Timing+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14_Timing+DigiFullTrigger_Timing_2023D17+RecoFullGlobal_Timing_2023D17+HARVESTFullGlobal_Timing_2023D17/step3_TTbar_14TeV_Timing+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14_Timing+DigiFullTrigger_Timing_2023D17+RecoFullGlobal_Timing_2023D17+HARVESTFullGlobal_Timing_2023D17.log

</pre>21234.0 step3
<pre>runTheMatrix-results/21234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D21_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D21+RecoFullGlobal_2023D21+HARVESTFullGlobal_2023D21/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D21_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D21+RecoFullGlobal_2023D21+HARVESTFullGlobal_2023D21.log

</pre>27434.0 step3
<pre>runTheMatrix-results/27434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D35+RecoFullGlobal_2023D35+HARVESTFullGlobal_2023D35/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D35+RecoFullGlobal_2023D35+HARVESTFullGlobal_2023D35.log

</pre>250202.181 step4
<pre>runTheMatrix-results/250202.181_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25/step4_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py TTbar_8TeV_TuneCUETP8M1_cfi  --conditions auto:run1_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot Realistic8TeVCollision : FAILED - time: date Fri Jan 18 01:23:05 2019-date Fri Jan 18 01:22:59 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:Fake2,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_Fake2 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --processName=HLTRECO --filein file:RelVal_Raw_Fake2_DATA.root --fileout file:RelVal_Raw_Fake2_DATA_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:24:08 2019-date Fri Jan 18 01:23:01 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_PRef_DATA.root --fileout file:RelVal_Raw_PRef_DATA_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:25:43 2019-date Fri Jan 18 01:23:06 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018_pp_on_AA --processName=HLTRECO --filein file:RelVal_Raw_HIon_DATA.root --fileout file:RelVal_Raw_HIon_DATA_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:28:44 2019-date Fri Jan 18 01:23:10 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_GRun_MC.root --fileout file:RelVal_Raw_GRun_MC_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:32:32 2019-date Fri Jan 18 01:23:14 2019 s - exit: 256
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc_l1stage1 --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_25ns : FAILED - time: date Fri Jan 18 01:23:22 2019-date Fri Jan 18 01:23:15 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_PRef_MC.root --fileout file:RelVal_Raw_PRef_MC_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:29:34 2019-date Fri Jan 18 01:23:17 2019 s - exit: 256
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_2016 : FAILED - time: date Fri Jan 18 01:23:25 2019-date Fri Jan 18 01:23:19 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_PIon_DATA.root --fileout file:RelVal_Raw_PIon_DATA_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:25:43 2019-date Fri Jan 18 01:23:24 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:Fake,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run1_mc_Fake --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --processName=HLTRECO --filein file:RelVal_Raw_Fake_MC.root --fileout file:RelVal_Raw_Fake_MC_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:28:44 2019-date Fri Jan 18 01:23:29 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018_pp_on_AA --processName=HLTRECO --filein file:RelVal_Raw_HIon_MC.root --fileout file:RelVal_Raw_HIon_MC_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:32:01 2019-date Fri Jan 18 01:23:51 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:Fake2,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake2 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --processName=HLTRECO --filein file:RelVal_Raw_Fake2_MC.root --fileout file:RelVal_Raw_Fake2_MC_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:29:31 2019-date Fri Jan 18 01:24:13 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:Fake1,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_Fake1 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_25ns --processName=HLTRECO --filein file:RelVal_Raw_Fake1_MC.root --fileout file:RelVal_Raw_Fake1_MC_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:32:21 2019-date Fri Jan 18 01:25:46 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:Fake,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run1_data_Fake --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --processName=HLTRECO --filein file:RelVal_Raw_Fake_DATA.root --fileout file:RelVal_Raw_Fake_DATA_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:28:52 2019-date Fri Jan 18 01:26:22 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_GRun_DATA.root --fileout file:RelVal_Raw_GRun_DATA_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:33:17 2019-date Fri Jan 18 01:28:47 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:Fake1,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_Fake1 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_25ns --processName=HLTRECO --filein file:RelVal_Raw_Fake1_DATA.root --fileout file:RelVal_Raw_Fake1_DATA_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:29:57 2019-date Fri Jan 18 01:28:50 2019 s - exit: 256
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --mc --scenario=pp -n 10 --conditions auto:run2_mc_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2018 --processName=HLTRECO --filein file:RelVal_Raw_PIon_MC.root --fileout file:RelVal_Raw_PIon_MC_HLT_RECO.root : FAILED - time: date Fri Jan 18 01:33:02 2019-date Fri Jan 18 01:28:55 2019 s - exit: 256

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

slava77 commented on RecoVertex/Configuration/python/RecoVertex_phase2_timing_cff.py:
```diff
@@ -0,0 +1,75 @@
+from SimTracker.TrackerHitAssociation.tpClusterProducer_cfi import tpClusterProducer
+from SimTracker.TrackAssociatorProducers.quickTrackAssociatorByHits_cfi import quickTrackAssociatorByHits
+from SimTracker.TrackAssociation.trackTimeValueMapProducer_cfi import trackTimeValueMapProducer
+from CommonTools.RecoAlgos.TOFPIDProducer_cfi import tofPID
```
the config parse errors were about this line

bendavid replied:
Yes I had missed this when renaming the file (and didn't notice because of an old .pyc file hanging around)

slava77 commented on RecoVertex/Configuration/python/RecoVertex_cff.py:
```diff
@@ -33,36 +33,47 @@
+from RecoVertex.Configuration.RecoVertex_phase2_timing_cff import *
 _phase2_tktiming_vertexrecoTask = cms.Task( vertexrecoTask.copy() ,
                                             tpClusterProducer ,
                                             quickTrackAssociatorByHits ,
```
remind me please why these truth matching modules are sitting in the RECO piece?

bendavid replied:
This is part of the "fast simulation" of the MTD used for the technical proposal, and which we want to keep around just a little while longer to facilitate direct comparisons.  After the TDR release it can be permanently removed.

slava77 commented on RecoVertex/Configuration/python/RecoVertex_phase2_timing_cff.py:
```diff
@@ -0,0 +1,75 @@
+from SimTracker.TrackerHitAssociation.tpClusterProducer_cfi import tpClusterProducer
+from SimTracker.TrackAssociatorProducers.quickTrackAssociatorByHits_cfi import quickTrackAssociatorByHits
+from SimTracker.TrackAssociation.trackTimeValueMapProducer_cfi import trackTimeValueMapProducer
+from CommonTools.RecoAlgos.TOFPIDProducer_cfi import tofPID
```
I'm beginning to somewhat regret my suggestion of factoring out the zoo of the phase2 vertexing from the default.
The problem with the 4 lines above is that they import something that is not used in this file.

It seems like there is a more clean way to do this and decouple significantly the phase2 part from the rest (see notes in the std reco file)

slava77 commented on RecoVertex/Configuration/python/RecoVertex_cff.py:
```diff
@@ -33,36 +33,47 @@
+#modifications for timing
+from RecoVertex.Configuration.RecoVertex_phase2_timing_cff import *
 _phase2_tktiming_vertexrecoTask = cms.Task( vertexrecoTask.copy() ,
                                             tpClusterProducer ,
```
I think that this ```_phase2_tktiming_vertexrecoTask``` (less the default ```vertexrecoTask.copy()```) together with ```_phase2_tktiming_layer_vertexrecoTask``` can be moved to the phase2 cff file; perhaps name them as ```tkTimingSpecific``` to indicate that they are just additions.

Then below all you need is to import these additions and do  ```phase2_timing.toReplaceWith(vertexrecoTask, cms.Task(vertexrecoTask.copy(),_phase2_tktimingSpecific_vertexrecoTask))```

bendavid replied:
The other modifications would still require importing unsortedOfflinePrimaryVertices4Dfastsim, etc though

bendavid replied:
And trying to split it this way I still get errors like:
"  File "/cvmfs/cms.cern.ch/slc6_amd64_gcc700/cms/cmssw/CMSSW_10_4_0_mtd3/python/FWCore/ParameterSet/SequenceVisitors.py", line 116, in enter
    self._names.append(self._moduleToName[node])
KeyError: cms.EDProducer("PrimaryVertexProducer",
"

slava77 replied:
that's fine,
in a similar way as the ```_phase2_tktimingSpecific_vertexrecoTask``` will have to be imported from the other file. One just wouldn't see how they are made

bendavid committed:
fix broken import

bendavid committed:
fix typo in default input collections

bendavid committed:
cleanup imports and modifications

cmsbuild commented:
The code-checks are being triggered in jenkins.

bendavid commented:
I made the imports explicit now to avoid the import *.  Let me know if you prefer it this way, or with "import *", or just going back to a single file...

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25628/8072

- This PR adds an extra 164KB to repository

- Found files with invalid states:
  - CommonTools/RecoAlgos/python/TOFPIDProducer_cfi.py:
    - Added: a0c1929358dfdfc8ff4a660084b0b4693f9a89e0
    - Modified: c834441618737eed29f1097cbfaf59430496c181, f487a29343e52b03fabc389b12228816f9a43647, cbda9a7e05f3e63a3b9fd4a8b1bd81f4fc6db948, b770cf63d1262306fa4e2bb237436d46ced5e583, f777755313b333cf770f408c3f05116ca7e6779f, 7810d5943e4b39736810665eaaf9d7f4d22f6007
    - Deleted: e1a553eff12d6d2f917ae4406f7c7d818fb2e7c2

cmsbuild commented:
Pull request #25628 was updated. @perrotta, @andrius-k, @kmaeshima, @schneiml, @kpedro88, @cmsbuild, @jfernan2, @slava77 can you please check and sign again.

slava77 commented:
On 1/17/19 6:49 PM, Josh Bendavid wrote:
> I made the imports explicit now to avoid the import *. Let me know if
> you prefer it this way, or with "import *", or just going back to a
> single file...
> 

explicit imports are fine

slava77 commented:
@cmsbuild please test workflow 20034.11

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: a4b2368bf90faabc1f0f8c65a5878ae24b27a245
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25628/32688/summary.html

cmsbuild commented:
Comparison job queued.

slava77 commented:
Some technicals for ttbar PU200

File sizes
```
117.0MiB ->   119.4MiB    +2.431MiB             2.1%     FEVT
 51.5MiB ->    52.7MiB    +1.174MiB             2.3%     RECO
 14.6MiB ->    14.8MiB     +0.214MiB             1.5%     AOD
```
These look acceptable.

Time is up by about 80 s/evt (the baseline is \~275 s/evt) 
- broken up roughly among the variety of PV reconstructions: 15.7 s unsortedOfflinePrimaryVertices3D and 29 s unsortedOfflinePrimaryVertices4DnoPID (both are new); +33.5 s (16 -> 59 s) in unsortedOfflinePrimaryVertices. 
- If I rescale the baseline to what was reported earlier in https://github.com/cms-sw/cmssw/pull/25628#issuecomment-455022098, the numbers are somewhat consistent.
- My guess is that @pmeridian @bendavid used a faster machine.

Considering that the zoo of PV recos will not stay in the future, this 30% increase in CPU per job is perhaps OK.
    - Note here that all the iter tracking adds up to \~60s compared to \~140s in the 4 PV variants. The new default PV is about the same as the total iter tracking time. I hope that this gets better eventually.

slava77 commented on RecoVertex/Configuration/python/RecoVertex_EventContent_cff.py:
```diff
@@ -45,4 +48,10 @@ def _phase2_tktiming_AddNewContent(mod):
 _phase2_tktiming_AddNewContent(RecoVertexRECO)
```
note that both phase2_timing and phase2_timing_layer are present in the MTD case
```
Era_Phase2C4_timing	= (Phase2C4, phase2_timing)
Phase2C4_timing_layer_bar	= (Phase2C4_timing, phase2_timing_layer, phase2_timing_layer_bar)
```

so, the event content will include both for the MTD TDR setup.
Is this actually desired?

slava77 replied:
if decoupled version is needed, IIRC the syntax is ```( phase2_timing & ~ phase2_timing_layer).toModify(```

bendavid replied:
Yes, this is intentional

bendavid commented:
The main motivation for swapping the default primary vertex collection was to easily propagate the vertices with timing to miniaod, quickly assessing whether there is another feasible way to do it.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25628/32688/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-25628/20034.11_TTbar_14TeV_Timing+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14_Timing+DigiFullTrigger_Timing_2023D17+RecoFullGlobal_Timing_2023D17+HARVESTFullGlobal_Timing_2023D17

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 708 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3096588
 - DQMHistoTests: Total failures: 2036
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3094355
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

bendavid committed:
revert to standard 3D vertices for default collections

bendavid committed:
add additional slimmed primary vertex collection to miniaod with timing

cmsbuild commented:
The code-checks are being triggered in jenkins.

bendavid commented:
After some discussion, we decided that the safest thing to do is to revert to the standard 3d vertices for the default collections. (4d vertices now populate "offlinePrimaryVertices4D" again)

To deal with the MINIAOD aspect, it turned out to be straightforward to simply add a second collection of slimmedOfflinePrimaryVertices to the MINIAOD (and this actually makes it easier to do timing vs no timing comparisons for some cases as well)

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25628/8077

- This PR adds an extra 168KB to repository

- Found files with invalid states:
  - CommonTools/RecoAlgos/python/TOFPIDProducer_cfi.py:
    - Added: a0c1929358dfdfc8ff4a660084b0b4693f9a89e0
    - Modified: c834441618737eed29f1097cbfaf59430496c181, f487a29343e52b03fabc389b12228816f9a43647, cbda9a7e05f3e63a3b9fd4a8b1bd81f4fc6db948, b770cf63d1262306fa4e2bb237436d46ced5e583, f777755313b333cf770f408c3f05116ca7e6779f, 7810d5943e4b39736810665eaaf9d7f4d22f6007
    - Deleted: e1a553eff12d6d2f917ae4406f7c7d818fb2e7c2

cmsbuild commented:
Pull request #25628 was updated. @perrotta, @andrius-k, @kmaeshima, @schneiml, @kpedro88, @cmsbuild, @jfernan2, @slava77, @santocch can you please check and sign again.

slava77 commented on PhysicsTools/PatAlgos/python/slimming/offlineSlimmedPrimaryVertices_cfi.py:
```diff
@@ -7,3 +7,7 @@
 )
+
+offlineSlimmedPrimaryVertices4D = cms.EDProducer("PATVertexSlimmer",
+    src = cms.InputTag("offlinePrimaryVertices4D"),
```
_cfi files should not be used to make multiple instances. Use _cff 
or add this into a separate offlineSlimmedPrimaryVertices4D_cfi.py.

Perhaps more importantly, is it OK in this case to skip rekey of the scores (ignored without the ```score``` argument)?

bendavid replied:
Yes, the re-keying is used for association between the packed candidates and the vertices, and a few other downstream things, which are still using only the 3d vertices for the moment now.

bendavid committed:
split 4D vertices into separate cfi file

cmsbuild commented:
The code-checks are being triggered in jenkins.

slava77 commented:
@cmsbuild please abort test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25628/8082

- This PR adds an extra 184KB to repository

- Found files with invalid states:
  - CommonTools/RecoAlgos/python/TOFPIDProducer_cfi.py:
    - Added: a0c1929358dfdfc8ff4a660084b0b4693f9a89e0
    - Modified: c834441618737eed29f1097cbfaf59430496c181, f487a29343e52b03fabc389b12228816f9a43647, cbda9a7e05f3e63a3b9fd4a8b1bd81f4fc6db948, b770cf63d1262306fa4e2bb237436d46ced5e583, f777755313b333cf770f408c3f05116ca7e6779f, 7810d5943e4b39736810665eaaf9d7f4d22f6007
    - Deleted: e1a553eff12d6d2f917ae4406f7c7d818fb2e7c2

cmsbuild commented:
Pull request #25628 was updated. @perrotta, @andrius-k, @kmaeshima, @schneiml, @kpedro88, @cmsbuild, @jfernan2, @slava77, @santocch can you please check and sign again.

slava77 commented:
@cmsbuild please test workflow 20034.11

I hope the both figures it out

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: bbff806ecce37866030b7916df5a80d7cf68fa77
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25628/32696/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25628/32696/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins/workspace/compare-root-files-short-matrix/results/JR-comparison/PR-25628/20034.11_TTbar_14TeV_Timing+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14_Timing+DigiFullTrigger_Timing_2023D17+RecoFullGlobal_Timing_2023D17+HARVESTFullGlobal_Timing_2023D17

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 34 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3097440
 - DQMHistoTests: Total failures: 329
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3096914
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

slava77 commented:
as expected, changes are now staying within the "MTD" vertices 
e.g. 
in the tenmu noPU (red is the baseline)
![all_origvssign1062-pass-bbff806_tenmuextendede2023d35wf27411p0c_log10recovertexs_offlineprimaryvertices4d__reco_obj_yerror](https://user-images.githubusercontent.com/4676718/51416217-61cfd700-1b2d-11e9-9944-3e3b0bec4418.png)
the baseline vertices are based on the fastsim timing, while the new ones have proper MTD simulation and PID applied (either PID or noPID in this case of 4D they look about the same).

with pileup  in TTbar PU200 200 events the same PV position error perspective shows an improvement 
![all_origvssign1062-pass-bbff806_ttbar14tev2023d35puwf27634p0c_log10recovertexs_offlineprimaryvertices4d__reco_obj_xerror](https://user-images.githubusercontent.com/4676718/51416318-cbe87c00-1b2d-11e9-8816-a701289eddd5.png)

The vertices show up as expected in the miniAOD
![all_mini_origvssign1062-pass-bbff806_ttbar14tev2023d35puwf27634p0c_log10recovertexs_offlineslimmedprimaryvertices4d__reco_obj_xerror](https://user-images.githubusercontent.com/4676718/51416374-179b2580-1b2e-11e9-81fa-b92ba9583ac5.png)

I'm not sure if PV4D DQM validation plots are particularly credible; still, here they are for TTbar PU200 (red is this PR; black is the baseline)
<img width="739" alt="wf27634-pass-bbff806_pv4d_dup_z" src="https://user-images.githubusercontent.com/4676718/51416704-7e6d0e80-1b2f-11e9-8f15-b94456c34469.png">
<img width="738" alt="wf27634-pass-bbff806_pv4d_eff_z" src="https://user-images.githubusercontent.com/4676718/51416707-82009580-1b2f-11e9-9fbb-8b986675bab7.png">
<img width="737" alt="wf27634-pass-bbff806_pv4d_fr_z" src="https://user-images.githubusercontent.com/4676718/51416714-862cb300-1b2f-11e9-9b8e-6cc68636ceee.png">
<img width="737" alt="wf27634-pass-bbff806_pv4d_mr_z" src="https://user-images.githubusercontent.com/4676718/51416719-89c03a00-1b2f-11e9-8dc8-02b16f0723ea.png">


CPU Time per event did not change significantly.

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/25628	bbff806 
- code changes are in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show differences only in D34 MTD workflow in the track extension MTD-related variables and in the 4D vertex
- local tests show somewhat expected behavior:
    - new modules for vertexing and tofPID are running : they increase reco time by about 30% (driven by the new kinds of PV vertexing: 4DnoPID and 4D [with tofPID])
    - new products show up in the outputs (RECO is increased by about 0.6%; AOD by about 1.4%; and miniAOD by about 3
    - some physics performance plots from these tests were posted in https://github.com/cms-sw/cmssw/pull/25628#issuecomment-455712072 and earlier in https://github.com/cms-sw/cmssw/pull/25628#issuecomment-455452273 (still applicable for track PID variables) . Considering a difference in presentation and a simplistic view, they appear to be consistent with what's provided with the PR description https://indico.cern.ch/event/783100/contributions/3273442/attachments/1780769/2896933/mtdvtxJan17-2019.pdf

slava77 commented:
@bendavid  please update the PR description to reflect the final state of the PR:

e.g. the following does not apply
> the timing eras has been reorganized such that the default "offlinePrimaryVertices" collection contains the 4D

pmeridian commented:
Wondering if here https://github.com/cms-sw/cmssw/blob/bbff806ecce37866030b7916df5a80d7cf68fa77/RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT_vect.cc#L130 we should std::isnan considering that this is what is returned when tracks has no time

kpedro88 commented:
+upgrade

andrius-k commented:
+1

fabiocos commented:
code-checks

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-25628/8127

- This PR adds an extra 180KB to repository

- Found files with invalid states:
  - CommonTools/RecoAlgos/python/TOFPIDProducer_cfi.py:
    - Added: a0c1929358dfdfc8ff4a660084b0b4693f9a89e0
    - Modified: c834441618737eed29f1097cbfaf59430496c181, f487a29343e52b03fabc389b12228816f9a43647, cbda9a7e05f3e63a3b9fd4a8b1bd81f4fc6db948, b770cf63d1262306fa4e2bb237436d46ced5e583, f777755313b333cf770f408c3f05116ca7e6779f, 7810d5943e4b39736810665eaaf9d7f4d22f6007
    - Deleted: e1a553eff12d6d2f917ae4406f7c7d818fb2e7c2

fabiocos commented on PhysicsTools/PatAlgos/python/slimming/slimming_cff.py:
```diff
@@ -56,3 +57,8 @@
+from Configuration.Eras.Modifier_phase2_timing_cff import phase2_timing
+_phase2_timing_slimmingTask = cms.Task(slimmingTask.copy(),
+                                       offlineSlimmedPrimaryVertices4D)
+phase2_timing.toReplaceWith(slimmingTask,_phase2_timing_slimmingTask)
```
@bendavid @slava77 given the other discussions in the review about phase2_timing vs phase2_timing_layer , is this really the desired behaviour?

bendavid replied:
Yes, this adds an additional slimmed primary vertex collection to the miniaod.  In the case of phase2_timing alone, this corresponds to the fastsim 4d vertices, in the case of phase2_timing_layer, this corresponds to fullsim 4d vertices.

fabiocos commented on PhysicsTools/PatAlgos/python/slimming/MicroEventContent_cff.py:
```diff
@@ -137,3 +137,7 @@
+
+_phase2_timing_extraCommands = ["keep *_offlineSlimmedPrimaryVertices4D_*_*"]
+from Configuration.Eras.Modifier_phase2_timing_cff import phase2_timing
+phase2_timing.toModify(MicroEventContentMC, outputCommands = MicroEventContentMC.outputCommands + _phase2_timing_extraCommands)
```
@bendavid @slava77 given the other discussions in the review about phase2_timing vs phase2_timing_layer , is this really the desired behaviour?

fabiocos commented:
+1

fabiocos commented:
@bendavid @pmeridian @slava77 @kpedro88 this PR is creating problems in the scenario D19, the old MTD simulation, as it crashes in the test 20434.0, see 

https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc6_amd64_gcc700/CMSSW_10_5_X_2019-01-22-2300/pyRelValMatrixLogs/run/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D19_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D19+RecoFullGlobal_2023D19+HARVESTFullGlobal_2023D19.log#/

where in the RECO step3 you get

----- Begin Fatal Exception 23-Jan-2019 05:23:18 CET-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 5 stream: 1
   [1] Running path 'validation_step6'
   [2] Prefetching for module CaloParticleValidation/'caloparticlevalidation'
   [3] Prefetching for module SimPFProducer/'simPFProducer'
   [4] Prefetching for module TOFPIDProducer/'tofPID'
   [5] Calling method for module PrimaryVertexProducer/'unsortedOfflinePrimaryVertices4DnoPID'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: edm::ValueMap&lt;float&gt;
Looking for module label: trackExtenderWithMTD
Looking for productInstanceName: generalTracksigmat0

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------

Do we really want to modify this scenario? My understanding is that @kpedro88 preferred to keep it frozen, waiting for the final new MTD simulation to become official to get rid of it

fabiocos commented:
@bendavid what is the purpose of touching the old simulation? Wouldn't be simpler just modify (phase2_timing_layer_tile | phase2_timing_layer_bar) ?

bendavid commented:
Looking at this now, the old scenario was not supposed to be modified here in the end (except for adding the new collection to miniaod which is generic), but since the configuration files for RecoVertex were heavily restructured, probably something was touched by mistake.

bendavid commented:
Ok, sorry, we're talking about two different things.
Indeed I was not aware of this, will rework the configs to fix it (but it makes things messy, because there will be phase2_timing_layer_tile | phase2_timing_layer_bar now in several places)

bendavid commented:
This actually would remove parts of the track and vertex validation which were already present in the old scenario (and using fastsim inputs).

I opened a new PR here which should really preserve the existing behaviour (except for adding the additional slimmed collection in miniaod, which would still be useful/desirable in case this is ever used)
https://github.com/cms-sw/cmssw/pull/25745

kpedro88 commented:
To combine with the `Phase2` Era to make top-level Eras for workflows. (Not essential, but possibly worthwhile for organization.)

makortel commented:
Ok, thanks.
